### PR TITLE
Added move_sprite example

### PR DESCRIPTION
**Pull request by jak6jak**
# Objective
There is no bevy example that shows how to transform a sprite. At least as its singular purpose. This creates an example of how to use transform.translate to move a sprite up and down.

## Solution

I created move_sprite example.

### CODE REVIEW

File: examples/2d/move_sprite.rs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(sprite_movement.system())
+        .run();
+}
+
+struct BevyLogo {
+    rising: bool,
```
Comment (DJMcNab) [inline examples/2d/move_sprite.rs:12]: We should use an enum for this.

File: Cargo.toml [hunk 2]
```diff
@@ -103,6 +103,10 @@ path = "examples/2d/contributors.rs"
 name = "mesh"
 path = "examples/2d/mesh.rs"
 
+[[example]]
+name="move_sprite"
+path="examples/2d/move_sprite.rs"
```
Comment (MinerSebas) [inline Cargo.toml:108]: The CI (https://github.com/bevyengine/bevy/pull/2391/checks?check_run_id=2909210823) doesn't seem to realize that example is present here. 🤔 
Perhaps the extra spaces are necessary here
```suggestion
[[example]]
name = "move_sprite"
path = "examples/2d/move_sprite.rs"
```

File: examples/2d/move_sprite.rs (hunk 1 above)
Comment (jak6jak) [reply examples/2d/move_sprite.rs:12]: I'm a bit new to rust and bevy ECS. can you explain why?

Comment (NathanSWard) [reply examples/2d/move_sprite.rs]: Rust enums provide an extra level of type safety and you can also name the variants to be more descriptive. 

effectively 
```rust
enum Movement {
    Rising, 
    Falling
}
```
and 
```rust
bool
```
are equivalent. 

However the `Movement` enum is more descriptive as to what the two states actually represent.

Comment (DJMcNab) [reply examples/2d/move_sprite.rs]: Especially if you call it something amusing like
`Logomotion`

**Review by alice-i-cecile — APPROVED** [review]
Simple, does what it says on the tin. Probably useful enough to beginners to be worth including.

File: examples/2d/move_sprite.rs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    asset_server: Res<AssetServer>,
+    mut materials: ResMut<Assets<ColorMaterial>>,
+) {
+    let texture_handle = asset_server.load("branding/icon.png");
+    commands.spawn_bundle(OrthographicCameraBundle::new_2d());
+    commands
+        .spawn_bundle(SpriteBundle {
+            material: materials.add(texture_handle.into()),
+            ..Default::default()
+        })
```
Comment (mockersf) [inline examples/2d/move_sprite.rs:27]: could you add a transform component to show how to place a sprite from the start?

**Review by mockersf — APPROVED** [review]

File: examples/2d/move_sprite.rs [hunk 4]
```diff
@@ -0,0 +1,45 @@
+use bevy::prelude::*;
+
+fn main() {
+    App::build()
+        .add_plugins(DefaultPlugins)
+        .add_startup_system(setup.system())
+        .add_system(sprite_movement.system())
```
Comment (NathanSWard) [inline examples/2d/move_sprite.rs:7]: on main the `.system()` syntax is no longer needed.
So we should probably also remove it here :)

**Review by NathanSWard — CHANGES_REQUESTED** [review]

**Comment by jak6jak** [comment]
okay I messed up trying to update my branch so that optional system was implemented. Can someone help?

**Comment by NathanSWard** [comment]
> okay I messed up trying to update my branch so that optional system was implemented. Can someone help?

If you can find the previous commit (before updating to main, reset to that branch)
```
git reset --hard <commit-hash>
```
Then when you update to main you want to rebase onto it.
For example I would do something like:
```
git fetch upstream
git rebase upstream/main
```
My remote repo for bevy (not my personal fork) is aliased as `upstream`.
So then I `rebase` on top of that main which will essentially pull all of your new commits off, apply main's commits, and then reapply yours.
See [here](https://git-scm.com/docs/git-rebase) to learn more about rebasing.

File: examples/2d/move_sprite.rs (hunk 4 above)
Comment (mockersf) [reply examples/2d/move_sprite.rs]: it can be done later, we will need a big pass on all examples anyway

**Comment by jak6jak** [comment]
I just decided to redo the pull request since I seemed to mess it up more :( https://github.com/bevyengine/bevy/pull/2414

**Pull request closed without merging**